### PR TITLE
Newly built Application

### PR DESCRIPTION
**Pull request by svelamuri26**
User Model completed.

### CODE REVIEW

File: src/main/java/com/example/NiRi/NiRiApplication.java [hunk 1]
```diff
@@ -3,40 +3,37 @@
...
+import org.springframework.context.annotation.ComponentScan;
+
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
 
 @SpringBootApplication
+@ComponentScan(basePackages = "com.example.NiRi")
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/NiRiApplication.java]: this is not needed in spring boot. The component scanning is done automatically

File: src/main/java/com/example/NiRi/NiRiApplication.java [hunk 2]
```diff
@@ -3,40 +3,37 @@
...
+			PreparedStatement pstmt = connection.prepareStatement(sql);{
 
-			try (PreparedStatement preparedStatement = connection.prepareStatement(insertSql)) {
-				preparedStatement.setString(1, name);
-				preparedStatement.setString(2, email);
-				preparedStatement.setString(3, password);
 
-				int affectedRows = preparedStatement.executeUpdate();
+				pstmt.setString(1, "Maria Haag");
+				pstmt.setString(2, "maria.haag@example.com");
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/NiRiApplication.java]: remove the comments and other logic you have around database from this file. In this class you should have just one static main method that is running the app and that's it. At least for now

File: src/main/java/com/example/NiRi/PasswordJWT.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package com.example.NiRi;
 
 
+
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/PasswordJWT.java]: extra line

File: src/main/java/com/example/NiRi/UserController.java [hunk 4]
```diff
@@ -12,6 +12,9 @@ public class UserController {
     @Autowired
     private UserService userService;
 
+    @Autowired
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/UserController.java]: can you use constructor autowiring please? https://stackoverflow.com/a/40620318

File: src/main/java/com/example/NiRi/UserController.java [hunk 5]
```diff
@@ -21,4 +24,24 @@ public User registerUser(@RequestBody User user) {
...
-}
+
+    @GetMapping("/getByEmail")
+    public User getUserByEmail(@RequestParam String email) {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new RuntimeException("User not found with email: " + email));
+    }
+
+    @PutMapping("/updateUser/{id}")
+    public User updateUserDetails(@PathVariable Long id, @RequestBody User updatedUser) {
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/UserController.java]: Can you move the logic from this function to your UserService

File: src/main/java/com/example/NiRi/UserService.java [hunk 6]
```diff
@@ -19,4 +19,26 @@ public User saveUser(User user) {
...
+    public User getUserByEmail(String email) {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new RuntimeException("User not found with email: " + email));
+    }
+
+    public User updateUserDetails(Long id, User updatedUser) {
+        User existingUser = userRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("User not found with id: " + id));
+
+
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/UserService.java]: extra line

File: src/main/java/com/example/NiRi/UserService.java [hunk 7]
```diff
@@ -19,4 +19,26 @@ public User saveUser(User user) {
...
+        existingUser.setName(updatedUser.getName());
+        existingUser.setEmail(updatedUser.getEmail());
+        existingUser.setPassword(updatedUser.getPassword());
+
+
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/UserService.java]: extra line

File: src/test/java/com/example/NiRi/UserServiceTest.java [hunk 8]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (faruka-startsteps) [inline src/test/java/com/example/NiRi/UserServiceTest.java]: extra line

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/NiRi/NiRiApplication.java (hunk 1 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/NiRiApplication.java]: I´ve removed it. Thnk you.

File: src/main/java/com/example/NiRi/NiRiApplication.java (hunk 2 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/NiRiApplication.java]: Removed the comments. Thnk you.

File: src/main/java/com/example/NiRi/PasswordJWT.java (hunk 3 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/PasswordJWT.java]: fixed it.

File: src/test/java/com/example/NiRi/UserServiceTest.java (hunk 8 above)
Comment (svelamuri26) [reply src/test/java/com/example/NiRi/UserServiceTest.java]: removed.

File: src/main/java/com/example/NiRi/UserService.java (hunk 7 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/UserService.java]: removed.

File: src/main/java/com/example/NiRi/UserService.java (hunk 6 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/UserService.java]: removed.

File: src/main/java/com/example/NiRi/UserController.java (hunk 5 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/UserController.java]: Done.

File: src/main/java/com/example/NiRi/UserController.java (hunk 4 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/UserController.java]: created it.

File: src/main/java/com/example/NiRi/ExceptionalHandling/InvalidMethodArgumentException.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    public InvalidMethodArgumentException(String errorKey, String message) {
+        super(message);
+        this.errorKey = errorKey;
+    }
+
+    public String getErrorKey() {
+        return null;
+    }
+}
+
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/ExceptionalHandling/InvalidMethodArgumentException.java]: Remove the empty lines please

File: src/main/java/com/example/NiRi/PasswordJWT.java [hunk 10]
```diff
@@ -1,23 +1,21 @@
...
 import io.jsonwebtoken.SignatureAlgorithm;
 import org.springframework.stereotype.Component;
-
 import java.util.Date;
 
 @Component
 public class PasswordJWT {
 
-    private static final String SECRET_KEY = "yourSecretKey"; // Replace with a secure key
+    private static final String SECRET_KEY = "aishwaryaniri221402";
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/PasswordJWT.java]: we should take the security key from application.properties

File: src/main/java/com/example/NiRi/Service/UserService.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    public User saveUser(User user) {
+        return userRepository.save(user);
+    }
+    public List<User> getAllUsers() {
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/Service/UserService.java]: add empty lines between functions

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/NiRi/ExceptionalHandling/InvalidMethodArgumentException.java (hunk 9 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/ExceptionalHandling/InvalidMethodArgumentException.java]: removed.

File: src/main/java/com/example/NiRi/PasswordJWT.java (hunk 10 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/PasswordJWT.java]: added in application.properties

File: src/main/java/com/example/NiRi/Service/UserService.java (hunk 11 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/Service/UserService.java]: added

File: src/main/java/com/example/NiRi/PasswordJWT.java (hunk 10 above)
Comment (faruka-startsteps) [reply src/main/java/com/example/NiRi/PasswordJWT.java]: I don't see any changes after these comments. Did you push your updates?

File: src/main/java/com/example/NiRi/PasswordJWT.java [hunk 12]
```diff
@@ -1,23 +1,22 @@
...
 import io.jsonwebtoken.SignatureAlgorithm;
 import org.springframework.stereotype.Component;
-
 import java.util.Date;
 
 @Component
 public class PasswordJWT {
 
-    private static final String SECRET_KEY = "yourSecretKey"; // Replace with a secure key
+    private static final String SECRET_KEY = "SECRET_KEY";
```
Comment (faruka-startsteps) [inline src/main/java/com/example/NiRi/PasswordJWT.java:12]: use the secret key from your application.properties file with @Value

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/NiRi/PasswordJWT.java (hunk 12 above)
Comment (svelamuri26) [reply src/main/java/com/example/NiRi/PasswordJWT.java:12]: Sure! I´l make the necessary changes.

**Pull request closed without merging**